horizomer/utils/tree.py

```diff
@@ -91,6 +144,66 @@ def intersect_trees(tree1, tree2):
     return (tree1_lap, tree2_lap)
 
 
+def collapse_short_branches(tree, cutoff):
+    """Collapse internal nodes with branch length below cutoff.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+        tree to search for nodes to collapse
+    cutoff : int or float
+        branch length cutoff under which node should be collapsed
+
+    Returns
+    -------
+    skbio.TreeNode
+        resulting tree with short branches collapsed
+
+    Notes
+    -------
+    Nodes without branch length are considered as having zero branch length and
+    will be collapsed.
+    """
+    tcopy = tree.copy()
+    nodes_to_collapse = []
+    for node in tcopy.non_tips():
+        if (node.length or 0.0) < cutoff:
+            nodes_to_collapse.append(node)
+    for node in nodes_to_collapse:
```

Comment:
Can we encounter recursive situations?

Reply:
Actually I did recursion in the very beginning. Then I thought a little bit and believed that there is no need for recursion. The first scan will collapse all short branches already. It is not possible that a tree after one round of collapse will have new branches that are below cutoff.

But please correct me if you think otherwise.

Reply:
@qiyunzhu definitely ok.  Recursion in Python is not natural, and is recommended to "unroll" the recursion (if I'm understanding this correctly).

Reply:
@mortonjt Thanks! In this case I think it is just logically not necessary to recurse or do any advanced tricks.